### DISCUSSION TITLE
E_IOC_LOOKUP_FAILED after latest update

### DISCUSSION THREAD

**Discussion by backspacerhino** [adonisjs/core#2152] [Help] [answered]
Hi,

I've updated everything to latest using
`npm i @adonisjs/core@alpha @adonisjs/lucid@alpha @adonisjs/auth@alpha @adonisjs/ace@alpha @adonisjs/mail@alpha @adonisjs/redis@alpha @adonisjs/fold@alpha @adonisjs/assembler@alpha`

But when I type `node ace serve -w` I get:

`E_IOC_LOOKUP_FAILED: Cannot resolve "Adonis/Core/Hash" namespace from the IoC Container`

for User.ts 

```
  @beforeSave()
  public static async hashPassword (user: User) {
    if (user.$dirty.password) {
      user.password = await Hash.make(user.password)
    }
  }

```
before update it worked completely fine.

My current package.json is like this:

```
{
  "name": "api-v5",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build",
    "start": "node ace serve --watch",
    "lint": "eslint . --ext=.ts"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^3.1.1",
    "@types/supertest": "^2.0.10",
    "adonis-preset-ts": "^1.0.4",
    "eslint": "^7.3.1",
    "eslint-plugin-adonis": "^1.0.13",
    "execa": "^5.0.0",
    "get-port": "^5.1.1",
    "japa": "^3.1.1",
    "pino-pretty": "^4.0.0",
    "supertest": "^4.0.2",
    "typescript": "^3.9.5",
    "youch": "^2.0.10",
    "youch-terminal": "^1.0.0"
  },
  "dependencies": {
    "@adonisjs/ace": "^8.0.0",
    "@adonisjs/auth": "^5.1.1",
    "@adonisjs/core": "^5.0.4-preview-rc-2.1",
    "@adonisjs/fold": "^7.0.9",
    "@adonisjs/lucid": "^10.0.0",
    "@adonisjs/mail": "^5.2.3",
    "@adonisjs/redis": "^5.0.9",
    "@adonisjs/view": "^3.1.0",
    "@types/jsdom": "^16.2.4",
    "@types/socket.io": "^2.1.8",
    "adonis5-scheduler": "^1.0.1",
    "apple-signin-auth": "^1.0.6",
    "form-data": "^3.0.0",
    "google-auth-library": "^6.0.5",
    "got": "^11.4.0",
    "jsdom": "^16.4.0",
    "mysql": "^2.18.1",
    "pg": "^8.3.3",
    "phc-argon2": "^1.0.10",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "socket.io": "^2.3.0",
    "source-map-support": "^0.5.19",
    "ws": "^7.3.1"
  }
}
```

**Comment by thetutlage** [comment, marked as answer]
Have you followed this blog post https://preview.adonisjs.com/blog/october-2020-release? Especially this point

<img width="750" alt="Screenshot 2021-01-20 at 8 12 34 PM" src="https://user-images.githubusercontent.com/1706381/105190324-e5c55b00-5b5b-11eb-8eb1-da8571fea43a.png">

**Reply by backspacerhino** [reply]
This did work, but now I am getting an error 

`"migration:run" command not found` when running `node ace migration:run` outside build folder. I didn't find any references that i must run it inside build folder, only this:

"All this confusion is a result of a stale build folder and you have to make sure that one process is always running to keep the build output up to date."

I do have another terminal running `node ace serve -w`
